### PR TITLE
GC tests

### PR DESCRIPTION
**Pull request by jmaxmarno**
working changes

### CODE REVIEW

File: tests/default.env [hunk 1]
```diff
@@ -1,4 +1,7 @@
 GB=mqtts://everyone:everyone@gb.wis2dev.io:8883
+GC_MQTT_BROKER=mqtts://everyone:everyone@gc.wis2dev.io:8883
 TRIGGER_BROKER=${TRIGGER_BROKER}
 GDC_API=https://gdc.wis2dev.io/collections/wis2-discovery-metadata
 GDC_CENTRE_ID=io-wis2dev-global-discovery-catalogue
+PROMETHEUS_HOST=https://prom.wis2dev.io/
```
Comment (tomkralidis) [inline tests/default.env:6]: Remove trailing slash and join in code?

File: tests/global_cache/gc_functional_tests.py [hunk 2]
```diff
@@ -17,17 +17,23 @@
...
+from shared_utils import mqtt_helpers, ab, prom_metrics
 
 
 # Connection strings for the development global broker and message generator
 # Access the environment variables
-mqtt_broker_out = os.getenv('MQTT_BROKER_OUT')
-mqtt_broker_in = os.getenv('MQTT_BROKER_IN')
-mqtt_broker_gc = os.getenv('MQTT_BROKER_GC')
+load_dotenv("../default.env")
+load_dotenv("../secrets.env.env")
```
Comment (tomkralidis) [inline tests/global_cache/gc_functional_tests.py:29]: `secrets.env.env` -> `secrets.env`

File: tests/shared_utils/prom_metrics.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    metrics = fetch_prometheus_metrics(metric_name, "", "", prometheus_baseurl, username, password)
+    for metric in metrics:
+        timestamp = float(metric['value'][1])
+        if not (current_time - expected_range <= timestamp <= current_time):
+            raise AssertionError(f"Metric {metric_name} timestamp {timestamp} is not within the expected range")
+
+
+if __name__ == '__main__':
+    # Example usage
+    import dotenv
```
Comment (tomkralidis) [inline tests/shared_utils/prom_metrics.py:41]: `from dotenv import load_dotenv` (for consistency)

**Review by tomkralidis — CHANGES_REQUESTED** [review]

File: tests/global_cache/gc_functional_tests.py (hunk 2 above)
Comment (jmaxmarno) [reply tests/global_cache/gc_functional_tests.py]: fixed

File: tests/shared_utils/prom_metrics.py (hunk 3 above)
Comment (jmaxmarno) [reply tests/shared_utils/prom_metrics.py]: fixed

**Pull request merged**